modules/internal/ChapelArray.chpl

```diff
@@ -3263,7 +3263,9 @@ module ChapelArray {
       if !a.stridable && b.stridable then
         compilerError("cannot assign from a stridable domain to an unstridable domain without an explicit cast");
 
-    a._instance.dsiAssignDomain(b, lhsPrivate=false);
+    // distributions/robust/arithmetic/resizing/resizeFromOtherLocale.chpl
+    // fails if this uses b._instance instead of b._value
+    a._value.dsiAssignDomain(b, lhsPrivate=false);
```

Comment:
@daviditen - I suspect that this reintroduces the bug that @benharsh fixed in PR #5171 but I'm not certain. I'm also not certain how to fix it. I'm worried that lhs._arrs will always be empty in chpl_assignDomainWithGetSetIndices if a was privatized. @benharsh - do you have any suggestions here?